### PR TITLE
Replace SharedDataProvider data source with couch

### PR DESCRIPTION
**Pull request by esoergel**
Just look up the forms directly in couch.  This could be slow, but it's much easier to guarantee correctness, and I don't think there are that many VHND forms (currently just under 2k on prod).

The VhndAvailabilityFluffPillow is very far behind, so looking in to what's wrong with it, it seemed like just replacing it with direct couch lookups would be more straightforward.

@czue @sravfeyn please review this carefully.  I've done a fair bit of due diligence on this one - done full text searches for all the stuff I removed, ran all the tests, inspected during runtime to make sure there's actual data, and verified that the relevant reports (CMR, BPR, HSR) all run.

Most of the code changes are deletions and changing the form properties referenced in the indicators to be the actual form properties (rather than the corresponding fluff calculator property).

### CODE REVIEW

File: custom/opm/reports.py [hunk 1]
```diff
@@ -416,31 +380,46 @@ class SharedDataProvider(object):
...
+        'stock_ors',
+        'stock_zntab',
+    ]
+
+    def get_all_vhnd_forms(self):
+        key = make_form_couch_key(DOMAIN, xmlns=VHND_XMLNS)
+        return get_db().view(
+            'reports_forms/all_forms',
+            startkey=key,
+            endkey=key+[{}],
```
Comment (dimagimon) [inline custom/opm/reports.py]: E226 missing whitespace around arithmetic operator

File: custom/opm/reports.py [hunk 2]
```diff
@@ -416,31 +380,46 @@ class SharedDataProvider(object):
...
+            reduce=False,
+            include_docs=True
+        ).all()
 
     @property
     @memoized
     def _service_dates(self):
```
Comment (esoergel) [inline custom/opm/reports.py]: Do a couch lookup to get all VHND forms, then assemble that same data structure based on the forms.

File: custom/opm/reports.py [hunk 3]
```diff
@@ -429,18 +416,19 @@ def _service_dates(self):
             }
         }
         """
```
Comment (esoergel) [inline custom/opm/reports.py]: Do a couch lookup to get all VHND forms, then assemble that same data structure based on the forms.

**Comment by sravfeyn** [comment]
Thanks Ethan. I have pulled this and tested on the local machine. Compared excel exports with Fluff and with Couch. Both of them agree. It looks good to me.

**Comment by czue** [comment]
@esoergel I don't think this is a good long term change. loading 2,000 (and linearly growing) forms for every time you view the report is still pretty bad. I think if you want to make this configurable via a flag and then set the flag temporarily to true that's fine, but it doesn't seem like we should delete all the existing logic as that will be superior long term

**Comment by dannyroberts** [comment]
Fun fact: I just stopped VhndAvailabilityFluffPillow on hqpillowtop0 after noticing that it was using **19GB of virtual memory**. Not sure how that is possible; perhaps it's storing every document it passes in memory somehow.

File: custom/opm/reports.py [hunk 4]
```diff
@@ -416,6 +380,29 @@ class SharedDataProvider(object):
     Data provider for report data that can be shared across rows in an instance
     of a report.
     """
+    vhnd_form_props = [
+        'attend_ANM',
+        'attend_ASHA',
+        'attend_cmg',
```
Comment (dimagimon) [inline custom/opm/reports.py]: E301 expected 1 blank line, found 0

**Comment by czue** [comment]
I chatted with @esoergel about this and he's going to make some changes to this and update it. if i'm not around you can assume i'm signed off

**Pull request merged**